overlay.nix

```diff
@@ -132,6 +132,10 @@ let
 
         dftbplus = super.python3.pkgs.toPythonApplication self.python3.pkgs.dftbplus;
 
+        dice = callPackage ./pkgs/by-name/dice/package.nix {
```

Comment:
You need to rebuild boost with an optimized MPI, otherwise you depend on two different version (optimized and unoptimized).


```
diff --git a/nixpkgs-opt.nix b/nixpkgs-opt.nix
index 3db064f..389b865 100644
--- a/nixpkgs-opt.nix
+++ b/nixpkgs-opt.nix
@@ -18,6 +18,7 @@ let
     avogadro2 = recallPackage avogadro2 {};
     arpack = recallPackage arpack {};
     arpack-mpi = recallPackage arpack { useMpi = true; };
+    boost-mpi = recallPackage boost { useMpi = true; mpi = self.mpi; };
     cp2k = recallPackage cp2k {};
     fftw = recallPackage fftw {};
     dkh = recallPackage dkh {};
diff --git a/overlay.nix b/overlay.nix
index b63c9d4..614dd2c 100644
--- a/overlay.nix
+++ b/overlay.nix
@@ -133,7 +133,7 @@ let
         dftbplus = super.python3.pkgs.toPythonApplication self.python3.pkgs.dftbplus;
 
         dice = callPackage ./pkgs/by-name/dice/package.nix {
-          boost = final.boost.override { useMpi = true; };
+          boost = self.boost-mpi;
         };
 
         dirac = callPackage ./pkgs/by-name/dirac/package.nix {
```

Reply:
Oh I wasn't aware that we have a recalled MPI. Good to know :sheep:

Reply:
This is a bit hidden unfortunately and a result of the optimized upstream package set.